### ISSUE TITLE
Sorting gets lost on page change

### ISSUE THREAD

**Issue by atufkas** [aghuddleston/Ext.ux.data.PagingStore#5]
Hi there! 

First of all: _Thank you for a great job sharing this fine work with the world!_ It's probably been a while for you since being in touch with this code - I'm aware that ExtJS 4 (and even 5) are a bit outdated by now.

I recently integrated the ExtJS 4.2 version of your PagingStore module into a legacy project I worked on from time to time over the past two years and discovered the following **bug**: When switching the page to be displayed with `loadPage(n)`, the current sorters are obviously getting lost and data is sorted as after initial loading with `loadRecords()`. After investigating some time (took a while to catch up with the roles of `allData` and `snapshot`) I think to have figured out what is causing this behavior and prepared a little patch as fix-branch which I want to kindly asked you for reviewing.

As far as I can see you're not explicitly testing this case within the Siesta test suite - you're only checking for a certain entry on the same page that is currently selected directly after applying some sorter. Unfortunately I wasn't able to get the test env up and running in a reasonable amount of time since dependencies on the CDN seem to broken. I tried with integrating from newer CDN urls (ExtJS 4.2) but that threw some errors _undefined variable: Ext_ or similar. I gave up for now und would continue here, including adding tests for the described case, if you generally approved my issue and solution.

Pull request follows immediately! 

Cheers & happy coding,
Matthias

**Comment by aghuddleston** [comment]
You are correct that it has been awhile since I've worked on this.  I'm currently not working w/ExtJS so I'm not even set up right now to try this out.  And it is hard to remember how 'allData' and 'snapshot' work together. 

One question for you though, you said ExtJS 4.2.  The master branch is Ext JS 4.1  Should this change be on the 4.2 branch?

**Comment by atufkas** [comment]
Ah yes, you're right - this should be on the 4.2 branch! 

As far as I can see the branches for 4.1 ad 4.2 don't differ in a way that would make the patch useless or unusable for either the other one. 

One change in the 4.2 branch adds a 1ms defer around a block not really belonging the issue (`@@ -244,25 +244,23 @@`) – for whatever reason. 

The second difference looks a bit like it _may_ affect the behavior of the issue but actually does not: Here the 4.2 branch just assigns `me.allData` instead of `me.snapshot` to `me.data` within the filter function – while this still looks a bit confusing to me it makes no difference on the fact that me.snapshot is used as a "fresh" basis before applying filters – which is desired but should e.g. happen here on a _sorted_ snapshot so that sorting is maintained.  (`@@ -466,9 +464,9 @@`).

Meaning: The patch should probably be applied to master/4.1 and 4.2 as well. The 5.0 branch though looks quite different - haven't checked this yet.

**Comment by aghuddleston** [comment]
I'm definitely ok w/fixing this, but would prefer to have the tests updated too.

**Comment by atufkas** [comment]
Okay, I'll soon take some time and fix the test suite plus integrate tests for the given case.
